tests/unit/projection-test.js

```diff
@@ -0,0 +1,1464 @@
+          },
+        }
+      });
+      store.push({
+        data: {
+          type: BOOK_EXCERPT_PROJECTION_CLASS_PATH,
+          id: FETCHED_PROJECTION_ID,
+          attributes: {},
+        },
+        included: [
+          {
+            type: BOOK_CLASS_PATH,
+            id: FETCHED_PROJECTION_ID,
+            attributes: {
+              title: `Mr. Popper's Penguins`
+            },
+            meta: {
+              partial: true,
+            }
+          },
+        ]
+      });
+    });
+
+    /*
+      Setup findRecord params for projection requests
+
+      remember:
+        the FETCHED_PROJECTION_ID is the unfetched base-record
+        the UNFETCHED_PROJECTION_ID is the already fetched base-record
+     */
+    findRecordCallCount = 0;
+    expectedFindRecordModelName = NORM_BOOK_EXCERPT_PROJECTION_CLASS_PATH;
+    trueFindRecordModelName = BOOK_EXCERPT_PROJECTION_CLASS_PATH;
+    expectedFindRecordId = UNFETCHED_PROJECTION_ID;
+
+    run(() => {
+      store.findRecord(BOOK_EXCERPT_PROJECTION_CLASS_PATH, FETCHED_PROJECTION_ID)
+        .then(model => {
+          assert.equal(get(model, 'id'), FETCHED_PROJECTION_ID, 'we retrieved the already fetched the model');
+          assert.equal(findRecordCallCount, 0, 'We did not re-fetch');
+        });
+    });
+
+    run(() => {
+      store.findRecord(BOOK_EXCERPT_PROJECTION_CLASS_PATH, UNFETCHED_PROJECTION_ID)
+        .then(model => {
+          assert.equal(get(model, 'id'), UNFETCHED_PROJECTION_ID, 'we fetched the model');
+          assert.equal(findRecordCallCount, 1, 'We made a single request');
+        });
+    });
+
+    /*
+      Setup findRecord params for base-record requests,
+
+      remember:
+        the FETCHED_PROJECTION_ID is the unfetched base-record
+        the UNFETCHED_PROJECTION_ID is the already fetched base-record
+    */
+    findRecordCallCount = 0;
+    expectedFindRecordModelName = NORM_BOOK_CLASS_PATH;
+    trueFindRecordModelName = BOOK_CLASS_PATH;
+    expectedFindRecordId = FETCHED_PROJECTION_ID;
+
+    run(() => {
+      store.findRecord(BOOK_CLASS_PATH, UNFETCHED_PROJECTION_ID)
+        .then(model => {
+          assert.equal(get(model, 'id'), UNFETCHED_PROJECTION_ID, 'we retrieved the already fetched the model');
+          assert.equal(findRecordCallCount, 0, 'We did not re-fetch');
+        });
+    });
+
+    run(() => {
+      store.findRecord(BOOK_CLASS_PATH, FETCHED_PROJECTION_ID)
+        .then(model => {
+          assert.equal(get(model, 'id'), FETCHED_PROJECTION_ID, 'we fetched the model');
+          assert.equal(findRecordCallCount, 1, 'We made a single request');
+        });
+    });
+  });
+
+  test('Projections proxy whitelisted attributes to a base-record', function(assert) {
+    const store = this.store();
+    const BOOK_ID = 'isbn:9780439708181';
+    const BOOK_TITLE = 'Adventures in Wonderland';
+    const BOOK_AUTHOR = 'Lewis Carroll';
+    const BOOK_DESCRIPTION = `Don't get rabbit holed!`;
```

Comment:
Please don't overly-DRY tests.  it makes the assertions harder to read.  There is no need to extract these strings.

Reply:
I find this style makes assertions much easier to read when dealing with the same contextual value in multiple contexts as we have here. Inlinining would hide away the "why" and "what" of the assertion.